src/test/javascript/spec/component/overview/course-lectures/course-lectures.component.spec.ts

```diff
@@ -0,0 +1,140 @@
+import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
+import { FaIconComponent } from '@fortawesome/angular-fontawesome';
+import { Lecture } from 'app/entities/lecture.model';
+import { Course } from 'app/entities/course.model';
+import { CourseLecturesComponent } from 'app/overview/course-lectures/course-lectures.component';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { CourseScoreCalculationService } from 'app/overview/course-score-calculation.service';
+import { ExerciseService } from 'app/exercises/shared/exercise/exercise.service';
+import { ArtemisDatePipe } from 'app/shared/pipes/artemis-date.pipe';
+import { MockTranslateService } from '../../../helpers/mocks/service/mock-translate.service';
+import * as moment from 'moment';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+@Component({ selector: 'jhi-course-lecture-row', template: '' })
+class CourseLectureRowStubComponent {
+    @Input() lecture: Lecture;
+    @Input() course: Course;
+    @Input() extendedLink = false;
+}
+
+class MockActivatedRoute {
+    parent: any;
+    params: any;
+
+    constructor(options: { parent?: any; params?: any }) {
+        this.parent = options.parent;
+        this.params = options.params;
+    }
+}
+
+const mockActivatedRoute = new MockActivatedRoute({
+    parent: new MockActivatedRoute({
+        params: of({ courseId: '1' }),
+    }),
+});
+describe('CourseLectures', () => {
+    let courseLecturesComponentFixture: ComponentFixture<CourseLecturesComponent>;
+    let courseLecturesComponent: CourseLecturesComponent;
+    let course: Course;
+    let lecture1: Lecture;
+    let lecture2: Lecture;
+    let lecture3: Lecture;
+
+    beforeEach(() => {
+        course = new Course();
+        course.id = 1;
+
+        const wednesdayBase = moment('18-03-2020', 'DD-MM-YYYY');
+        const wednesdayBefore = moment('11-03-2020', 'DD-MM-YYYY');
+        const wednesdayAfter = moment('25-03-2020', 'DD-MM-YYYY');
+
+        lecture1 = new Lecture();
+        lecture1.id = 1;
+        lecture1.startDate = wednesdayBase;
+        lecture2 = new Lecture();
+        lecture2.id = 2;
+        lecture2.startDate = wednesdayBefore;
+        lecture3 = new Lecture();
+        lecture3.id = 3;
+        lecture3.startDate = wednesdayAfter;
+        course.lectures = [lecture1, lecture2, lecture3];
+
+        TestBed.configureTestingModule({
+            imports: [NgbDropdownModule],
+            declarations: [
+                CourseLecturesComponent,
+                CourseLectureRowStubComponent,
+                MockPipe(TranslatePipe),
+                MockPipe(ArtemisDatePipe),
+                MockComponent(SidePanelComponent),
+                MockComponent(FaIconComponent),
+                MockDirective(JhiTranslateDirective),
+            ],
+            providers: [
+                MockProvider(CourseManagementService, {
+                    getCourseUpdates: () => {
```

Comment:
Cool way mock the service, I really like it. Maybe we can have a code guideline for this, so that we write all our service mocks like this :)